src/python/base/memoize.py

```diff
@@ -39,11 +39,22 @@ class FifoInMemory(object):
 
   def __init__(self, capacity):
     self.capacity = capacity
-    self.cache = collections.OrderedDict()
     self.lock = threading.Lock()
+    self._cache = None
+
+  @property
+  def cache(self):
+    """Get the cache backing. None may be returned."""
+    if self._cache is None:
+      self._cache = collections.OrderedDict()
+
+    return self._cache
 
   def put(self, key, value):
     """Put (key, value) into cache."""
+    if self.cache is None:
```

Comment:
How can it be None if it is initialized to collections.OrderedDict() in cache property function.

Reply:
It can be None if it's overridden by a child class.

Reply:
Ok.